main.py

```diff
@@ -0,0 +1,77 @@
+import random
+import re
+
+def random_names_generator():
+    def input_name():
+        """
+        Function that takes names input and check if input is valid
+        Returns
+        ----
+        name_list : list
+        """
```

Comment:
Nice use of comments here tells exactly what the function does and the return values